server/utils/date.js

```diff
@@ -0,0 +1,62 @@
+const { updateDatesBySubscriptionId } = require('../api/actions/subscriptions');
+const { addDays, addMonths } = require('../../client/src/utils/sharedDateUtils');
+
+const updateNextDueDate = async (dueDate, subscriptionUuid) => {
+  const { frequency, occurence, dates } = dueDate;
+  const todaysDate = new Date(new Date().setHours(0, 0, 0, 0)).toISOString().substring(0, 10);
+  let lateDueDate;
+  let newDueDate;
+  if (frequency === 'daily') {
+    newDueDate = new Date(dates[0]);
+    if (newDueDate < new Date(todaysDate)) {
+      lateDueDate = newDueDate;
+
+      while (newDueDate < new Date(todaysDate)) {
+        newDueDate = new Date(addDays(newDueDate, parseInt(occurence, 10)));
+      }
+    }
+  }
+
+  const newDatesObject = await dates.map((day) => {
```

Comment:
this can/should just be part of the if/else above to simplify logic. additionally, we can just combine some of the logic to make this all a bit easier to read and follow:

```javascript
  let lateDueDate;

  const formatDate = (frequency) => {
    let newDueDate;
    if (frequency === 'daily') {
      newDueDate = new Date(dates[0]);
      if (newDueDate < new Date(todaysDate)) {
        lateDueDate = newDueDate;

        while (newDueDate < new Date(todaysDate)) {
          newDueDate = new Date(addDays(newDueDate, parseInt(occurence, 10)));
        }
      }
      return [newDueDate.toISOString().substring(0, 10)];
    } else {
      const newDatesObject = dates.map((day) => {
        const date = new Date(day);
        if (date < new Date(todaysDate)) {
          lateDueDate = date.toISOString().substring(0, 10);

          if (frequency === 'yearly') {
            return addMonths(date, 12);
          } else if (frequency === 'monthly') {
            return addMonths(date, parseInt(occurence, 10));
          } else if (frequency === 'weekly') {
            return new Date(addDays(date, occurence * 7)).toISOString().substring(0, 10);
          }
        }
        return day;
      });
      return newDatesObject;
    }
  }

  const body = {
    dueDate: {
      frequency, 
      occurence,
      dates: formatDate(frequency),
      lateDueDate,
    },
    subscriptionUuid
  };
```